tests/integration/mounts_nodev_nosuid_noexec.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	# Create a ro fuse-sshfs mount; skip the test if it's not working.
+	local sshfs="sshfs
+		-o UserKnownHostsFile=/dev/null
+		-o StrictHostKeyChecking=no
+		-o PasswordAuthentication=no"
+
+	DIR="$BATS_RUN_TMPDIR/fuse-sshfs"
+	mkdir -p "$DIR"
+
+	if ! $sshfs -o nodev,nosuid,noexec rootless@localhost: "$DIR"; then
+		skip "test requires working sshfs mounts"
+	fi
+
+	setup_busybox
+	update_config '.process.args = ["/bin/echo", "Hello World"]'
+}
+
+function teardown() {
+	# New distros (Fedora 35) do not have fusermount installed
+	# as a dependency of fuse-sshfs, and good ol' umount works.
+	fusermount -u "$DIR" || umount "$DIR"
+
+	teardown_bundle
+}
+
+@test "runc run [dev,exec,suid bind mount of a nodev,nosuid,noexec fuse sshfs mount]" {
+	# The "sync" option is used to trigger a remount with the below options.
+	# It serves no further purpose. Otherwise only a bind mount without
+	# applying the below options will be done.
+	update_config '	  .mounts += [{
+					type: "bind",
+					source: "'"$DIR"'",
+					destination: "/mnt",
+					options: ["dev", "suid", "exec", "rprivate", "rbind", "sync"]
```

Comment:
Can we test just using `"rbind", "rprivate"` too? We hit that in https://github.com/opencontainers/runc/issues/3770 with COS 

cc @vinayakankugoyal

Reply:
Unfortunately this is not sufficient. It will not trigger a remount.

Reply:
Oh, because of this: https://github.com/opencontainers/runc/blob/5726682966d27c3fcf00089cc994c5566d3ce910/libcontainer/rootfs_linux.go#L471-L473 ?

Then this is solving a real problem for you? You happen to hit it and the flags that were used included some that trigger a remount?

Reply:
You could try `"rbind", "rprivate", "sync"`. In order to trigger a remount an additional flag *different* to `unix.MS_REC|unix.MS_REMOUNT|unix.MS_BIND` must be present. `"dev", "suid", "exec"` do not meet this as they are parsed to set no flag (their value is `true`):

https://github.com/opencontainers/runc/blob/5726682966d27c3fcf00089cc994c5566d3ce910/libcontainer/specconv/spec_linux.go#L74-L77

Reply:
Yes, I understand that. But the only time I hit this in the wild was without such an option being set. It doesn't seem like crun needs that either.

With this patch, are you fixing an issue you encounter in a real world scenario?

Reply:
Once I patched runc with the patch here it worked fine.

Reply:
Cool, that is great, then! :)

Reply:
> It would be great if it also fixed that case, as some maintainers were suggesting to do that too here: [#3801 (reply in thread)](https://github.com/opencontainers/runc/discussions/3801#discussioncomment-5535407)

It fixes the situation mentioned in  [#3801 (reply in thread)](https://github.com/opencontainers/runc/discussions/3801#discussioncomment-5535407) as the issue hit me with using k3s rootless which means that we were running in a separate user namespace.

Reply:
Hmm, no, you said it won't work for that case a few messages above: https://github.com/opencontainers/runc/pull/3805#discussion_r1159606627.

It will work for the k3s case, that happens to have other "special" flags, but not for the case where those flags are not used, that is what is discussed in that discussion link (https://github.com/opencontainers/runc/discussions/3801#discussioncomment-5535407)

Reply:
It does solve #3801 as in #3801 they use `rbind` and `ro`. `ro` will trigger a remount and this remount does fail with the current code as we do the remount without `nodev` and `nosuid` which are set for the source mount point `/run` in this case. With the code in this PR it will succeed. Just mounting with `rbind`, `rprivate` is not enough as it will not trigger a remount. It will just work with the current code and the code in this PR. My understanding is that the remount is needed to apply additional mount options as just doing a bind mount will not apply them to the bind mounted target. If we want to apply them we need to remount, but the remount will fail in rootless cases if we do not add `ro`, `nodev`, `nosuid`, `noexec` to the remount if they are present on the source of the bind mount.

Reply:
Oh, great, it seems containerd is using the ro flag, but I mixed them up in my initial comment: https://github.com/containerd/containerd/pull/8336/files#diff-8429def96328014c5ad61aa0611c15e050161fc24da5a6934bdedc55e69b6815L120

Can we add a test for a mount with rbind and ro too, then? So we don't regress in this other real world case either?

Reply:
What should be the mount options of the source? I understand for the target you want to use `rbind` and `ro`.

Reply:
@rpluem-vf probably nodev,nosuid,noexec are enough. But it would be great if @vinayakankugoyal shares what options the COS mount of `/var` has.

Reply:
So `nodev`, `nosuid`, `noexec` on the source and `rbind`, `ro` on the destination (pending @vinayakankugoyal confirmation)?

Reply:
Yes :)

Reply:
How about 06f65bf8b44151bdadf0dd07ef4cc67434140247 that is now added?

Reply:
SGTM